runtime/src/bank.rs

```diff
@@ -3401,4 +3400,16 @@ mod tests {
         let last_ts = bank1.last_vote_sync.load(Ordering::Relaxed);
         assert_eq!(last_ts, 1);
     }
+
+    #[test]
+    fn test_hash_internal_state_unchanged() {
```

Comment:
@sagar-solana Err, late comment, but maybe outdated test name? should be `test_hash_internal_state_changed`? (i.e. `unchanged` => `changed`)

Reply:
Never too late :)
It's funny, I think both names work. I went with unchanged because the internal state is unchanged but the test is testing that the hash changes lol

Reply:
Oh, now I got it!

Hmm, sure funny thing. Then how about adding a comment to the assertion like this?:

```rust
+        // hashes should be different even for banks with same internal state.
         assert_ne!(bank0_hash, bank1_hash); 
```